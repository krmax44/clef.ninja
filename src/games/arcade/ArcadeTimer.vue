<template>
	<div class="timer" :class="{ away: state !== 'playing' }">
		<div
			class="timer-inner"
			:class="{ hurry: remainingTime <= 10 }"
			:style="{ width: `${(remainingTime / totalTime) * 100}%` }"
		/>
	</div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
	props: {
		state: {
			type: String,
			required: true
		},
		remainingTime: {
			type: Number,
			required: true
		},
		totalTime: {
			type: Number,
			required: false,
			default: 60
		}
	}
});
</script>

<style lang="postcss" scoped>
.timer {
	transition: all 0.5s;
	@apply fixed inset-x-0 top-0 bg-gray-200;

	&.away {
		opacity: 0;
		transform: translateY(-8px);
	}

	&-inner {
		height: 8px;
		transition: all 1s linear;
		@apply bg-brand-blue;

		&.hurry {
			@apply bg-brand-red;
		}
	}
}
</style>
