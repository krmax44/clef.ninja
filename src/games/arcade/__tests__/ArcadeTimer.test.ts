import { mount } from '@vue/test-utils';
import ArcadeTimer from '../ArcadeTimer.vue';

describe('ArcadeTimer', () => {
	it('renders correctly', () => {
		const wrapper = mount(ArcadeTimer, {
			propsData: {
				state: 'playing',
				remainingTime: 5
			}
		});

		expect(wrapper.element).toMatchSnapshot();
	});
});
