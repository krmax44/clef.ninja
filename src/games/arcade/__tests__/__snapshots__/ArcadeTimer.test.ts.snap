// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArcadeTimer renders correctly 1`] = `
<div
  class="timer"
>
  <div
    class="timer-inner hurry"
    style="width: 8.333333333333332%;"
  />
</div>
`;
